docs/data/toolpad/core/components/app-provider/app-provider.md

```diff
@@ -6,20 +6,90 @@ components: AppProvider
 
 # App Provider
 
-<p class="description">The app provider component provides the necessary context for a Toolpad application, such as routing, navigation and theming.</p>
+<p class="description">The app provider component provides the necessary context to easily set up a Toolpad application.</p>
+
+By wrapping an application at the root level with an `AppProvider` component, many of Toolpad's features (such as routing, navigation and theming) can be automatically enabled to their fullest extent, abstracting away complexity and helping you focus on the details that matter.
+
+It is not mandatory that every application is wrapped in an `AppProvider`, but it is highly recommended for most apps that use Toolpad.
 
 ## Basic
 
-Wrap the whole application with the `AppProvider` component to enable many of Toolpad's features.
+Wrap an application page with the `AppProvider` component.
+
+Ideally, the `AppProvider` should wrap every page in the application, therefore in most projects it should be imported and placed in the file that defines a **shared layout** for all pages.
+
+In the following (purely demonstrative) example, an `AppProvider` component wrapping the page provides it with a default theme, and a `DashboardLayout` placed inside it gets its navigation and routing features automatically set based on the props passed to the `AppProvider`.
```

Comment:
Not sure it brings much to the table 🙂 
```suggestion
In the following example, an `AppProvider` component wrapping the page provides it with a default theme, and a `DashboardLayout` placed inside it gets its navigation and routing features automatically set based on the props passed to the `AppProvider`.
```

Reply:
Yeah this one was because you mentioned before that this example with a `DashboardLayout` directly inside an `AppProvider` could be misleading as it's not how most users would/should use them, so I wasn't even sure if I should include this example. I'm fine with removing that part!